determined-ai/determined
# feat: Logic of different modes for webhook

## Ticket

[MD-479]


## Description

Part of workload alerting project.
ERD link: https://hpe-aiatscale.atlassian.net/wiki/spaces/ENGINEERIN/pages/1666809868/Workload+Alerting+ERD

Expand the experiment config -> integrations -> webhooks
```
integrations:
  webhooks:
    webhook_name:
      - test-webhook
```

Below is a table for which group of experiments can trigger which mode of webhooks.
workspace \ mode     |                             Specific                         |                       Workspace
| :------------------- | :-------------------------------------: | :------------------------------------: |
workspace defined    |  match workspace && match config      |      all experiments within workspace
workspace null           |                      match config                       |                     all experiments


## Test Plan

Unit tests and e2e tests should be able to cover it. 
To test from WebUI:

> - Create a webhook with workspace ID
> - Triggering the webhook with experiments within that workspace should work
> - The webhook should not be triggered with experiments outside of that workspace

> - Create a webhook with mode specific
> - experiment should only trigger that webhook with matching config


## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.




[MD-479]: https://hpe-aiatscale.atlassian.net/browse/MD-479?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
